### PR TITLE
Pet profile route+controller

### PR DESCRIPTION
**Pull request by konradkop**
### What this PR does (required):
- Created the Route and Controller for Pet Profiles
- 
### Screenshots / Videos (required):
petprofile

### CODE REVIEW

**Comment by moffatethan** [comment]
I've updated the base of the branch to the pet model branch, this reduces the file sizes and keeps things more concise 😄

File: server/controllers/petProfile.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    const values = {
+        _id,
+        petName,
+        ownerID,
+        petType,
+        petAge,
+        petStatus,
+     } = req.body;
+
+     for(let prop in values) if(!values[prop]) delete values[prop];//it will remove fields who are undefined or null
```
Comment (moffatethan) [inline server/controllers/petProfile.js:36]: This isn't very readable, could we make it a bit more readable?

**Review by moffatethan — CHANGES_REQUESTED** [review]
Overall good, let's just do that readability change 😄 👍🏻

**Comment by konradkop** [comment]
> Another thing, I would like to see middleware validation on the incoming data. We should be validating all incoming data we are accepting from users to ensure reliability in our codebase.

OK, I've added the jwt authentication middlewere that was in the codebase as well as some type checks

**Review by moffatethan — APPROVED** [review]
LGTM

**Pull request merged**